### ISSUE TITLE
Empty parameter lists aren't ignored during comparison

### ISSUE THREAD

**Issue by nox** [hyperium/mime#101]
Comparing `"text/event-stream;".parse().unwrap()` with `TEXT_EVENT_STREAM` yields `false`. That seems like a bug in this crate to me.

**Comment by nox** [comment]
If that's not a bug, I wonder if there should be some easy way to compare MIME types in a way that ignores parameters, because that's a thing that the HTML spec wants to do, at least for `text/event-stream`. Maybe `PartialEq<(Name, Name)>` for `Mime`?

**Comment by seanmonstar** [comment]
I think the empty parameter list should have been chopped off at parsing, so that's a bug.

For comparing, I think there's two ways:

1. Use `type_()` and `subtype()`.
2. With that's on master, the second type could be a `MediaRange`.

  ```rust
  let range = MediaRange::from(TEXT_EVENT_STREAM);
  range.matches(&media_type);
  ```

**Comment by seanmonstar** [comment]
Fixed and published in 0.3.13.
